### ISSUE TITLE
Matching name and id on form element should not fail duplicate-id-aria

### ISSUE THREAD

**Issue by grace-snow** [microsoft/accessibility-insights-web#2285]
## Describe the bug

On a fast pass web test, the accessibility insights chrome extension is flagging the following markup as **duplicate-id-aria** and giving the warning "Fix the following: Document has multiple elements referenced with ARIA with the same id attribute: sortCode". However, there is only one instance of sortCode as an id on the page.

```
<label for="sortCode">
	Bank sort code
</label>
<input id="sortCode" type="tel" inputmode="numeric" name="sortCode" class="content__input-text number-only" size="6" maxlength="6" title="Enter 6 digit sort code" required value="<%=agreement.getBank_sort_code() != null ? agreement.getBank_sort_code() : " "%>" />
<div id="sortCode-err" class="error hidden" role="alert" aria-live="assertive"></div>
````

WCAG21 says this html is valid, accessible html: "The `id` attribute may have the same value as the `name` attribute, but both must be provided, and the id must be unique in the Web page. "  https://www.w3.org/WAI/WCAG21/Techniques/html/H44.html

**To Reproduce**

1. Insert the above html onto a webpage
2. Run 'fast pass' in the accessibility insights extension
3. See the error message

**Note:** 
* This was tested by changing the error message id so it doesn't begin with the same text (`sortCode-err` was changed to `sort-code-err`) and the fail remained on re-running fast pass.
* This was tested by changing the input name to be different to the ID (`name="sortCode"` was changed to `name="sort-code"`) and the fail was gone on re-running fast pass  

## Expected behavior

Name and id on an input can be the same, no `duplicate-id-aria` fail flagged.

## Screenshots

![image](https://user-images.githubusercontent.com/10631991/75863249-12fc0400-5df8-11ea-8ece-efcf84d039a5.png)
![image](https://user-images.githubusercontent.com/10631991/75863335-332bc300-5df8-11ea-89fd-2995fc596b62.png)
![image](https://user-images.githubusercontent.com/10631991/75863384-463e9300-5df8-11ea-98c7-e69011ed409b.png)

## Context (please complete the following information)

- OS Name & Version: Windows 10 Pro (1803)
- Browser Version: Google Chrome 78.0.3904.87 (Official Build) (64-bit)

## Are you willing to submit a PR?

Yes, but I am unsure how to help.

## Did you search for similar existing issues?

Yes, but didn't find a matching report

**Comment by pownkel** [comment]
Hi @grace-snow, unfortunately I wasn't able to reproduce the bug with the steps listed (I tried with [this codepen](https://codepen.io/smoralesd/pen/XWbemmX)). I noticed that you seem to be using some sort of template engine for the required value, so I tried both replacing that string with an arbitrary value and removing it altogether, but it didn't seem to change anything significant.

Since it doesn't seem like that code snippet alone is causing the issue, could you provide some more context to help us reproduce this? A codepen or a sample webpage would be helpful.
